### PR TITLE
[REFACTOR] Deeplink Processing Stabilization.

### PR DESCRIPTION
**Pull request by doyoonkim3312**
## 📝 Summary (개요)
- Modify Deeplink Processing logic to stabilize deeplink received via ImplicitIntent/PendingIntent
- Apply missing Manifest Rule to WidgetConfigurationActivity to enhance UX.
## 🔗 Related Issue (관련 이슈)
- Resolves _(if applicable)_ #
- Jira Ticket: [KAN-105]

## 🛠 Type of Change (변경 사항)
- [ ] `FEAT`: New feature (새로운 기능)
- [x] `FIX`: Bug fix (버그 수정)
- [x] `REFACTOR`: Code refactoring (no functional change) (코드 리팩토링)
- [ ] `DESIGN`: UI/UX changes (디자인 변경)
- [ ] `!HOTFIX`: Critical fix (치명적인 버그 수정)
- [x] `CHORE`: Build/Config/CI (빌드/설정/CI)

## ✨ Key Changes (핵심 변경 사항)
  - Resolve identified Deeplink-related edge case. (OS launch Task from History)
  - Stabilize current Deeplink processing logic (utilize Channel to ensure correct intent processing while preventing duplicated navigation.
  - Modify Manifest Rule for WidgetConfigurationActivity to enhance UX

## 📸 Screenshots / Video (스크린샷 또는 동영상)
| Before (이전) | After (이후) |
|:---:|:---:|
| <video src="https://github.com/user-attachments/assets/e9fbdd6d-8ae1-4b96-94af-b76c223aed1a" width="300" /> | <video src="https://github.com/user-attachments/assets/eb3a1ddf-9db4-4d56-a49a-ac041375df45" width="300" /> |

## 🧪 Test Plan (테스트 계획)
- [x] **Device**: Android Emulator (Google Pixel 9 Pro, SDK 36)
- [x] **Scenario**: (Deeplink Processing Test) Initiate Pending Intent via Push Notification while app is in Foreground/Background -> Requested Deeplink is correctly processed.
- [x] **Scenario**: (Edge Case Scenario) Launch Application via either Pending Intent or Implicit Intent -> Press back button until app is closed -> Press system Overview button -> Re-enter the app -> Check whether app is correctly navigate user to HomeScreen. (Previous Deeplink is not being processed)

## ✅ Checklist (체크리스트)
- [x] My code follows the style guidelines of this project (코드 스타일을 준수했습니다).
- [x] I have performed a self-review of my own code (스스로 코드를 검토했습니다).
- [x] I have commented my code, particularly in hard-to-understand areas (이해하기 어려운 부분에 주석을 작성했습니다).

[KAN-105]: https://knutice.atlassian.net/browse/KAN-105?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/src/main/java/com/doyoonkim/knutice/MainActivity.kt [hunk 1]
```diff
@@ -123,6 +126,21 @@ class MainActivity : ComponentActivity() {
...
+                    intentChannel.receiveAsFlow().collect { intent ->
+                        isDeeplinkProcessing.value = true
+                        DeeplinkHandler.processDeeplink(intent) { host, destination ->
+                            // TODO: Revert Analytics Logging later.
+                            navController.navigate(destination) {
+                                launchSingleTop = true
+                            }
+                        }
+                        isDeeplinkProcessing.value = false
+                    }
```
[the remark this answers is not in the document]
Comment (doyoonkim3312) [reply app/src/main/java/com/doyoonkim/knutice/MainActivity.kt:131]: 1. Current Analytics logging is too general. It will be restored in later version with more detailed Logging feature.
2. Resolved ([fc8c0b9](https://github.com/KNUTICE/KNUTICE-Android/pull/100/commits/fc8c0b98a84465b02a08134e34c45dc29af76bc3))
3. Resolved ([fc8c0b9](https://github.com/KNUTICE/KNUTICE-Android/pull/100/commits/fc8c0b98a84465b02a08134e34c45dc29af76bc3))

**Review by jeongHunE — APPROVED** [review]
